source/agora/consensus/protocol/Nominator.d

```diff
@@ -769,16 +769,34 @@ extern(D):
         const cur_height = this.ledger.height();
         log.trace("Received BLOCK SIG {} from node {} for block {}",
                     block_sig.signature, block_sig.utxo, block_sig.height);
-        if (block_sig.height > cur_height)
-            return BlockHeader.init;
-        const block = this.ledger.getBlocksFrom(Height(block_sig.height)).front;
-        if (!this.collectBlockSignature(block_sig, block.hashFull()))
-            return BlockHeader.init;
-        BlockHeader header = block.header.clone();
-        this.updateMultiSignature(header);
-        const updated_sig = header;
-        this.ledger.updateBlockMultiSig(updated_sig);
-        return updated_sig;
+
+        const pendingBlockSig = (block_sig.height == this.pending_block.header.height);
+        log.dbg("{}: Signature is for pending block #{} still to be externalized", __FUNCTION__, this.pending_block.header.height);
+
+        if (!pendingBlockSig && block_sig.height > cur_height) // Too early for this signature
+        {
+            early_sigs.put(block_sig);
```

Comment:
Hold on, if someone sends a fake signature for 100 blocks in the future, we will always store it, won't we ?

Reply:
Good point, I will add a limit of 3 blocks.